src/pages/commons/use-local-storage.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+import { useState } from 'react';
+import { load, save } from '../../common/localStorage';
+
+export function useLocalStorage<T>(key: string, defaultValue?: T) {
+  const [value, setValue] = useState(() => load(key) ?? defaultValue);
+
+  function handleValueChange(newValue: T) {
```

Comment:
I would wrap this with useCallback

Reply:
I wouldn't unless you have a benchmark with numbers